src/transforms/DummyMainFunction.cpp

```diff
@@ -0,0 +1,262 @@
+/** Insert dummy main function if one does not exist */
+
+#include <string.h>
+#include "llvm/Pass.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/IR/Value.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Type.h"
+#include "llvm/IR/GlobalValue.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Demangle/Demangle.h"
+
+#include "seadsa/CompleteCallGraph.hh"
+#include "seadsa/InitializePasses.hh"
+
+//#include "seahorn/Support/SeaDebug.h"
+#include "boost/format.hpp"
+
+static llvm::cl::opt<std::string>
+EntryPoint("entry-point",
+        llvm::cl::desc ("Entry point if main does not exist"),
+        llvm::cl::init (""));
+
+namespace previrt {
+    namespace transforms {
+
+
+        using namespace llvm;
+
+
+        class DummyMainFunction : public ModulePass
+        {
+            DenseMap<const Type*, Constant*> m_ndfn;
+
+            std::vector<std::string> parseArgs(std::string str){
```

Comment:
Please do not do this. LLVM allows you  to define `llvm::cl::list<string>` (https://llvm.org/docs/CommandLine.html#parsing-a-list-of-options) so you don't need to painfully parse the string

Reply:
I’ll use this